File: backend/AIE_backend-main/AIO_API/Entities/CampaignSession.cs
using AIO_API.Entities.Campaigns;
using AIO_API.Interfaces;

namespace AIO_API.Entities
{
    public class CampaignSession : ICampaignSession
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public DateTime CreateDate { get; set; }
        public string? Description { get; set; }
        public int CampaignId { get; set; }
        public Campaign Campaign { get; set; }

        public void Create(int campaignId)
        {
            CampaignId = campaignId;
            CreateDate = DateTime.UtcNow;
        }
    }
}

File: frontend/lib/features/campaigns/domain/campaign_session.dart
class CampaignSession {
  final int id;
  final int campaignId;
  final String title;
  final String? description;
  final DateTime createDate;

  const CampaignSession({
    required this.id,
    required this.campaignId,
    required this.title,
    required this.createDate,
    this.description,
  });

  factory CampaignSession.fromJson(Map<String, dynamic> json) {
    return CampaignSession(
      id: json['id'] as int,
      campaignId: json['campaignId'] as int,
      title: (json['title'] ?? '') as String,
      description: json['description'] as String?,
      createDate: DateTime.parse(json['createDate'] as String),
    );
  }
}
